R/Record.R

```diff
@@ -94,6 +94,127 @@ Record <- R6::R6Class( # nolint object_name_linter
           )
         )
       }
+    },
+    print = function(style = TRUE) {
+      cli::cat_line(self$to_string(style))
+    },
+    to_string = function(style = FALSE) {
+      field_order <- c(
+        # Simple fields
+        "uid",
+        "handle",
+        "name",
+        "root",
+        "n",
+        "dtype",
+        "unit",
+        "description",
+        "synonyms",
+        "key",
+        "suffix",
+        "type",
+        "size",
+        "source_code",
+        "registry",
+        "hash",
+        "n_objects",
+        "n_observations",
+        "started_at",
+        "finished_at",
+        "is_consecutive",
+        "reference",
+        "reference_type",
+        "visibility",
+        "version",
+        "is_latest",
+        "region",
+        "instance_uid",
+        "created_at",
+        "updated_at",
+
+        # Relational fields
+        "created_by",
+        "storage",
+        "transform",
+        "transforms",
+        "run",
+        "report",
+        "environment",
+        "meta_artifact",
+        "ulabels",
+        "predecessors",
+        "successors",
+        "runs",
+        "parent",
+        "parents",
+        "children",
+        "output_artifacts",
+        "input_artifacts",
+        "output_collections",
+        "input_collections",
+        "input_of_runs",
+        "feature_sets",
+        "collections",
+        "features",
+        "artifacts",
+        "values",
+        "created_transforms",
+        "created_runs",
+        "created_artifacts"
+      )
+
+      column_names <- map(private$.registry$get_fields(), "column_name") |>
+        unlist()
+      column_names <- names(column_names)
+
+      # Reorder names according to set order
+      field_names <- intersect(field_order, column_names)
+      # Make sure any unknown names are included
+      field_names <- c(field_names, sort(setdiff(column_names, field_names)))
+
+      field_strings <- purrr::map_chr(field_names, function(.name) {
+        # Get value, handling missing/empty/inaccessible fields
+        value <- try(self[[.name]], silent = TRUE)
+        if (inherits(value, "try-error")) {
+          return(NA_character_)
+        }
+
+
+        if (inherits(value, c("Record", "R6"))) {
+          value <- try(value$id, silent = TRUE)
+
+          if (inherits(value, "try-error")) {
+            return(NA_character_)
+          }
+
+          .name <- paste0(.name, "_id")
+        }
```

Comment:
Oof! This means all of the linked fields will be queried from the API. If we want to print the ids of related fields, it might be better to call

```R
        private$.api$get_record(
          module_name = field$module_name,
          registry_name = field$registry_name,
          id_or_uid = private$.data[["uid"]],
          include_foreign_keys = TRUE
        )
```

instead, as this only performs one extra API call.

Reply:
It took me a while but I think I worked out what you meant. This is actually cleaner anyway.

Reply:
:+1: